src/main/java/com/gluonhq/RunMojo.java

```diff
@@ -0,0 +1,136 @@
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+@Mojo(name = "run", requiresDependencyResolution = ResolutionScope.RUNTIME)
+@Execute(phase = LifecyclePhase.PROCESS_CLASSES)
+public class RunMojo extends NativeBaseMojo {
+
+    private static final String POM_XML = "pom.xml";
+    private static final String RUN_POM_XML = "runPom.xml";
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        try {
+            createSubstrateDispatcher();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new MojoExecutionException("Error creating Substrate Dispatcher", e);
+        }
+
+        final InvocationRequest invocationRequest = new DefaultInvocationRequest();
+        invocationRequest.setProfiles(project.getActiveProfiles().stream()
+                .map(Profile::getId)
+                .collect(Collectors.toList()));
+        invocationRequest.setProperties(session.getRequest().getUserProperties());
+
+        // 1. Read pom
+        File pomFile = new File(POM_XML);
+        // 2. Create temporary pom file
+        File runPomFile = new File(RUN_POM_XML);
+        try (InputStream is = new FileInputStream(POM_XML)) {
+            // 3. Create model from current pom
+            Model model = new MavenXpp3Reader().read(is);
+
+            model.getBuild().getPlugins().stream()
+                    .filter(p -> p.getGroupId().equalsIgnoreCase("org.openjfx") &&
+                            p.getArtifactId().equalsIgnoreCase("javafx-maven-plugin"))
+                    .findFirst()
+                    .orElseThrow(() -> new MojoExecutionException("No JavaFX plugin found"));
+            
+            // 4. Check for Attach Dependencies and if Desktop is supported, add the classifier
+            model.getDependencies().stream()
+                    .filter(p -> p.getGroupId().equalsIgnoreCase(AttachArtifactResolver.DEPENDENCY_GROUP))
+                    .filter(p -> Arrays.stream(AttachService.values())
+                            .filter(AttachService::isDesktopSupported)
+                            .anyMatch(attach -> attach.getServiceName().equalsIgnoreCase(p.getArtifactId()))
+                    )
+                    .forEach(p -> p.setClassifier("desktop"));
+
+            // 5. Serialize new pom
+            try (OutputStream os = new FileOutputStream(runPomFile)) {
+                new MavenXpp3Writer().write(os, model);
+            }
+        } catch (Exception e) {
+            if (runPomFile.exists()) {
+                runPomFile.delete();
+            }
+            throw new MojoExecutionException("Error reading pom", e);
+        }
+
+        if (!project.getActiveProfiles().isEmpty()) {
+            final String activeProfiles = project.getActiveProfiles().stream().map(Profile::getId).collect(Collectors.joining(","));
```

Comment:
What do you mean by "filter out `desktop` ?

Reply:
`desktop` is a valid profile, so it shouldn't trigger a warning if you run `mvn -Pdesktop client:run`. Of course, there is no need to add it, as it is usually activated by default, but there are cases where it is explicitly added.
Therefore you need to remove it from `activeProfiles`.

Reply:
`desktop` is just a profile id that we use in our gluon samples repository. It can be changed at any point in time or a user can decide to use some other profile id other than `desktop` in their projects using `client-maven-plugin`.

Reply:
you are right, to do it properly  we should filter out the profile with target = "host"... but even that might be customised with a different property name.
Up to you...

Reply:
May be just leave it as it is for now?

Reply:
Ok